llama_hub/llama_packs/arize_phoenix_query_engine/base.py

```diff
@@ -0,0 +1,90 @@
+"""
+Arize-Phoenix LlamaPack.
+"""
+
+from types import ModuleType
+from typing import TYPE_CHECKING, Any, Dict, List
+
+from llama_index import set_global_handler
+from llama_index.indices.vector_store import VectorStoreIndex
+from llama_index.llama_pack.base import BaseLlamaPack
+from llama_index.schema import TextNode
+
+if TYPE_CHECKING:
+    from phoenix import Session as PhoenixSession
+
+
+class ArizePhoenixQueryEnginePack(BaseLlamaPack):
+    """
+    The Arize-Phoenix LlamaPack show how to instrument your LlamaIndex query
+    engine with tracing. It launches Phoenix in the background, builds an index
+    over an input list of nodes, and instantiates and instruments a query engine
+    over that index so that trace data from each query is sent to Phoenix.
+
+    Note: Using this LlamaPack requires that your OpenAI API key is set via the
+    OPENAI_API_KEY environment variable.
+    """
+
+    def __init__(
+        self,
+        nodes: List[TextNode],
+        **kwargs: Any,
+    ) -> None:
+        """
+        Initializes a new instance of ArizePhoenixQueryEnginePack.
+
+        Args:
+            nodes (List[TextNode]): An input list of nodes over which the index
+            will be built.
+        """
+        set_global_handler("arize_phoenix")
+        phoenix = _import_phoenix()
+        self._session: PhoenixSession = phoenix.launch_app()
```

Comment:
import and launch_app should precede set_global_handler, otherwise `HttpExporter` will [warn](https://github.com/run-llama/llama_index/blob/c61a146a096883644019edb2dac040483301fa22/llama_index/callbacks/arize_phoenix_callback.py#L15) if session is not active.
```suggestion
        phoenix = _import_phoenix()
        self._session: PhoenixSession = phoenix.launch_app()
        set_global_handler("arize_phoenix")
```

Reply:
addressed in df4e929